src/analyses/ai.h

```diff
@@ -36,7 +36,8 @@ class ai_baset
   typedef ai_domain_baset statet;
   typedef goto_programt::const_targett locationt;
 
-  ai_baset()
+  explicit ai_baset(std::unique_ptr<ai_domain_factory_baset> &&df)
```

Comment:
Hm, so this means now it's impossible to implement an `ait` without a factory? I think this was possible before.

Reply:
It was possible before and remains possible.  `ait<domainT>` has a constructor that creates a suitable factory (using `domainT::domainT()`).  This PR should be source compatible with the previous code base.